src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -125,71 +126,120 @@ void AppHost::LastTabClosed(const winrt::Windows::Foundation::IInspectable& /*se
-            nonClient = Viewport::FromDimensions({ _currentWidth,
-                                                   _currentHeight })
-                            .ToRect();
+            // If the title bar is out-of-screen, we set the initial position to
+            // the top left corner of the nearest monitor
+            proposedRect.left = monitorInfo.rcWork.left;
+            proposedRect.top = monitorInfo.rcWork.top;
         }
-    }
 
-    const auto adjustedHeight = nonClient.bottom - nonClient.top;
-    const auto adjustedWidth = nonClient.right - nonClient.left;
+        auto initialSize = _app.GetLaunchDimensions(dpix);
+
+        const short _currentWidth = Utils::ClampToShortMax(
+            static_cast<long>(ceil(initialSize.X)), 1);
+        const short _currentHeight = Utils::ClampToShortMax(
+            static_cast<long>(ceil(initialSize.Y)), 1);
+
+        // Create a RECT from our requested client size
+        auto nonClient = Viewport::FromDimensions({ _currentWidth,
+                                                    _currentHeight })
+                             .ToRect();
+
+        // Get the size of a window we'd need to host that client rect. This will
+        // add the titlebar space.
+        if (_useNonClientArea)
+        {
+            // If we're in NC tabs mode, do the math ourselves. Get the margins
+            // we're using for the window - this will include the size of the
+            // titlebar content.
+            const auto pNcWindow = static_cast<NonClientIslandWindow*>(_window.get());
+            const MARGINS margins = pNcWindow->GetFrameMargins();
+            nonClient.left = 0;
+            nonClient.top = 0;
+            nonClient.right = margins.cxLeftWidth + nonClient.right + margins.cxRightWidth;
+            nonClient.bottom = margins.cyTopHeight + nonClient.bottom + margins.cyBottomHeight;
+        }
+        else
+        {
+            bool succeeded = AdjustWindowRectExForDpi(&nonClient, WS_OVERLAPPEDWINDOW, false, 0, dpix);
+            if (!succeeded)
+            {
+                // If we failed to get the correct window size for whatever reason, log
+                // the error and go on. We'll use whatever the control proposed as the
+                // size of our window, which will be at least close.
+                LOG_LAST_ERROR();
+                nonClient = Viewport::FromDimensions({ _currentWidth,
+                                                       _currentHeight })
+                                .ToRect();
+            }
+
+            // For client island scenario, there is an invisible border of 8 pixels.
+            // We need to remove this border to guarantee the left edge of the window
+            // coincides with the screen
+            const auto pCWindow = static_cast<IslandWindow*>(_window.get());
+            const RECT frame = pCWindow->GetFrameBorderMargins(dpix);
+            proposedRect.left -= (frame.left * -1);

```

Comment:
"left minus left*-1" sounds like "left + left"; can this be simplified?

Reply:
Sorry for this stupid mistake, changed